src/Microsoft.Framework.Runtime/ApplicationEnvironment.cs

```diff
@@ -49,5 +49,41 @@ public FrameworkName RuntimeFramework
         {
             get { return _targetFramework; }
         }
+
+#if DNX451
+        public object GetData(string name)
+        {
+            return AppDomain.CurrentDomain.GetData(name);
+        }
+
+        public void SetData(string name, object value)
+        {
+            AppDomain.CurrentDomain.SetData(name, value);
+        }
+#else
+        // NOTE(anurse): ConcurrentDictionary seems overkill here. This data is rarely used, and a global lock seems safer.
+        private object _lock = new object();
+        private Dictionary<string, object> _appGlobalData = new Dictionary<string, object>();
```

Comment:
I'll mirror what Get/SetData on AppDomain do (which I don't know and will test first :))